app/presenters/tree_node_builder.rb

```diff
@@ -62,108 +69,126 @@ def build
+    "MiqServer"                  => -> { miq_server_node },
+    "MiqTemplate"                => -> { generic_node(object.name, "currentstate-#{object.normalized_state.downcase}.png") },
+    "MiqAlert"                   => -> { generic_node(object.description, "miq_alert.png") },
+    "MiqAction"                  => -> { miq_action_node },
+    "MiqEventDefinition"         => -> { generic_node(object.description, "event-#{object.name}.png") },
+    "MiqGroup"                   => -> { generic_node(object.name, "group.png") },
     # Following line has dynatree workaround, add class to allow clicking on bold portion of title.
-    when MiqPolicy            then miq_policy_node
-    when MiqPolicySet         then generic_node(object.description, "policy_profile#{object.active? ? "" : "_inactive"}.png")
-    when MiqUserRole          then generic_node(object.name, "miq_user_role.png")
-    when OrchestrationTemplateCfn then generic_node(object.name, "orchestration_template_cfn.png")
-    when OrchestrationTemplateHot then generic_node(object.name, "orchestration_template_hot.png")
-    when OrchestrationTemplateAzure then generic_node(object.name, "orchestration_template_azure.png")
-    when OrchestrationTemplateVnfd then generic_node(object.name, "orchestration_template_vnfd.png")
-    when ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate then generic_node(object.name, "orchestration_template_vapp.png")
-    when PxeImage             then generic_node(object.name, object.default_for_windows ? "win32service.png" : "pxeimage.png")
-    when WindowsImage         then generic_node(object.name, "os-windows_generic.png")
-    when PxeImageType         then generic_node(object.name, "pxeimagetype.png")
-    when PxeServer            then generic_node(object.name, "pxeserver.png")
-    when ScanItemSet          then generic_node(object.name, "scan_item_set.png")
-    when Service              then generic_node(object.name, object.picture ? "/pictures/#{object.picture.basename}" : "service.png")
-    when ServiceResource      then generic_node(object.resource_name, object.resource_type == "VmOrTemplate" ? "vm.png" : "service_template.png")
-    when ServerRole           then server_role_node(object)
-    when ServiceTemplate      then service_template_node
-    when ServiceTemplateCatalog then service_template_catalog_node
-    when Snapshot             then snapshot_node
-    when Storage              then generic_node(object.name, "storage.png")
-    when Switch               then generic_node(object.name,
+    "MiqPolicy"                  => -> { miq_policy_node },
+    "MiqPolicySet"               => -> { generic_node(object.description, "policy_profile#{object.active? ? "" : "_inactive"}.png") },
+    "MiqUserRole"                => -> { generic_node(object.name, "miq_user_role.png") },
+    "OrchestrationTemplateCfn"   => -> { generic_node(object.name, "orchestration_template_cfn.png") },
```

Comment:
possibly make these a single node (for base class) and lookup the png name in a hash / case statement like we do in Ems?